### PR TITLE
Opencrvs form flashes

### PR DESCRIPTION
**Pull request by wilrona**
This PR enables :
- Prevent the form from resetting when you are on the informant's details page.
- Saves changes to informant information when FIELD_AGENT is selected as the informant.

### CODE REVIEW

File: packages/client/src/components/form/FormFieldGenerator.tsx [hunk 1]
```diff
@@ -397,7 +397,7 @@ const GeneratedInputField = React.memo<GeneratedInputFieldProps>(
             label={fieldDefinition.label}
             name={fieldDefinition.name}
             value={String(value)}
-            selected={(value as string) === checkedValue}
+            selected={Boolean(value) === Boolean(checkedValue)}
```
Comment (rikukissa) [inline packages/client/src/components/form/FormFieldGenerator.tsx:400]: Do we still need this?

File: packages/client/src/views/RegisterForm/RegisterForm.tsx [hunk 2]
```diff
@@ -724,7 +724,7 @@ class RegisterFormView extends React.Component<FullProps, State> {
         data: {
           ...modifiedDeclaration.data,
           informant: {
-            informantType: modifiedDeclaration.data.informant.informantType
+            ...modifiedDeclaration.data.informant
```
Comment (rikukissa) [inline packages/client/src/views/RegisterForm/RegisterForm.tsx:727]: Why was this needed?

**Comment by Zangetsu101** [comment]
@rikukissa @wilrona This flashing issue was introduced by a interim solution which we wanted to address in this ticket: https://github.com/opencrvs/opencrvs-core/issues/5987

File: packages/client/src/views/RegisterForm/RegisterForm.tsx (hunk 2 above)
Comment (wilrona) [reply packages/client/src/views/RegisterForm/RegisterForm.tsx:727]: the phone number field has a dynamicInitialValue but when the change is made, it is not persisted.

File: packages/client/src/views/RegisterForm/RegisterForm.tsx [hunk 3]
```diff
@@ -744,9 +744,9 @@ class RegisterFormView extends React.Component<FullProps, State> {
       this.props.modifyDeclaration(modifiedDeclaration)
       // this is to forcefully remount the component
       // to reset the initial values of formik
-      this.setState({
-        formFieldKey: `${this.props.activeSection.id}-${declaration.data.informant.informantType}`
-      })
+      // this.setState({
+      //   formFieldKey: `${this.props.activeSection.id}-${declaration.data.informant.informantType}`
+      // })
```
Comment (rikukissa) [inline packages/client/src/views/RegisterForm/RegisterForm.tsx:749]:
```suggestion
```

File: packages/client/src/views/RegisterForm/RegisterForm.tsx (hunk 2 above)
Comment (rikukissa) [reply packages/client/src/views/RegisterForm/RegisterForm.tsx:727]: I was looking at this and started wondering, isn't 

```ts
      modifiedDeclaration = {
        ...modifiedDeclaration,
        data: {
          ...modifiedDeclaration.data,
          informant: {
            ...modifiedDeclaration.data.informant
          }
        }
      }
```
the same as 

```ts
      modifiedDeclaration = {
        ...modifiedDeclaration,
        data: {
          ...modifiedDeclaration.data,
          informant: modifiedDeclaration.data.informant
        }
      }
```

which is the same as 

```ts
      modifiedDeclaration = {
        ...modifiedDeclaration,
        data: modifiedDeclaration.data
      }
```

which is the same as no changes at all right? 🤔

**Comment by Zangetsu101** [comment]
@rikukissa I reckon this would require #6149 to work correctly right? In that case would be nice to cherry pick this to core

**Review by rikukissa — APPROVED** [review]

**Review by rikukissa — APPROVED** [review]
